CC/Sounder/receiver.cc

```diff
@@ -164,7 +165,43 @@ void Receiver::loopRecv(int tid, int core_id, SampleBuffer* rx_buffer)
 
     printf("receiver thread %d has %d radios\n", tid, radio_end - radio_start);
 
+    // prepare BS beacon in host buffer
+    std::vector<void*> beaconbuff(2);
+    std::vector<void*> zeros(2);
+    zeros[0] = calloc(config_->sampsPerSymbol, sizeof(int16_t) * 2);
+    zeros[1] = calloc(config_->sampsPerSymbol, sizeof(int16_t) * 2);
+    beaconbuff[0] = config_->beacon_ci16.data();
+    beaconbuff[1] = zeros[0];
+
+    size_t frameTimeLen = config_->sampsPerSymbol * config_->symbolsPerFrame;
+    size_t txFrameDelta = std::ceil(TIME_DELTA / (1e3 * frameTimeLen / config_->rate));
+    size_t txTimeDelta = txFrameDelta * frameTimeLen;
+
+    long long rxTimeBs(0);
+    long long txTimeBs(0);
+
+    // read rx_offset to align the FPGA time of the BS
+    // by performing dummy readStream()
+    std::vector<std::complex<int16_t>> syncbuffBs(config_->sampsPerSymbol * config_->symbolsPerFrame, 0);
+    std::vector<void*> syncrxbuffBs(2);
+    syncrxbuffBs[0] = syncbuffBs.data();
+
+    std::cout << "Sync BS host and FPGA timestamp..." << std::endl;
+    baseRadioSet_->radioRx(0, syncrxbuffBs.data(), config_->sampsPerSymbol, rxTimeBs);
+
+    // schedule the first beacon in the future
+    txTimeBs = rxTimeBs + config_->sampsPerSymbol * config_->symbolsPerFrame * 10;
+    baseRadioSet_->radioTx(0, beaconbuff.data(), 2, txTimeBs);
+    long long bsInitRxOffset = txTimeBs - rxTimeBs;
+    for (int it = 0; it < std::floor(bsInitRxOffset / config_->sampsPerSymbol); it++) {
+        baseRadioSet_->radioRx(0, syncrxbuffBs.data(), config_->sampsPerSymbol, rxTimeBs);
+    }
```

Comment:
It looks like this part breaks the sounder when run on Faros. Can we condition this block of code to UHD case only?

Reply:
Done.

Have you tested this code on Faros and make sure the "default" Iris mode still runs smoothly?

Reply:
Yes, I got a segfault in this line.